cedar-policy-core/src/entities.rs

```diff
@@ -338,10 +339,14 @@ fn create_entity_map(es: impl Iterator<Item = Entity>) -> Result<HashMap<EntityU
 impl IntoIterator for Entities {
     type Item = Entity;
 
-    type IntoIter = hash_map::IntoValues<EntityUID, Entity>;
+    type IntoIter = std::vec::IntoIter<Entity>;
 
     fn into_iter(self) -> Self::IntoIter {
-        self.entities.into_values()
+        self.entities
+            .into_values()
+            .map(Arc::unwrap_or_clone)
+            .collect::<Vec<Entity>>()
```

Comment:
Having to `.collect()` here seems like a potential performance problem as well ... is it possible to avoid it with a different choice of `type IntoIter` above, so that the implementation here can just be `self.entities.into_values().map(Arc::unwrap_or_clone)`?  or alternately, should we change the `type Item` to be `Arc<Entity>`?

Reply:
Definitely, I'll check that out.

Reply:
Updated in 11e70bd. I changed the IntoIter type to reflect the .map call as the return type. This gave another speed boost as is.